### PR TITLE
Migrate certain device routes to be higher up in paths

### PR DESCRIPTION
**Pull request by oestrich**
Since we have unique identifiers across the entire NH, we can skip having the org and the product in the route. Role validation will happen in the route actions for now, and as the rest of the controller is filled out we can figure out a better way to load these and validate.

### CODE REVIEW

File: lib/nerves_hub_web/controllers/api/device_controller.ex [hunk 1]
```diff
@@ -13,8 +14,7 @@ defmodule NervesHubWeb.API.DeviceController do
 
   plug(:validate_role, [org: :delete] when action in [:delete])
   plug(:validate_role, [org: :write] when action in [:create, :update])
-  plug(:validate_role, [org: :write] when action in [:reboot, :reconnect, :code, :upgrade])
```
Comment (jjcarstens) [inline lib/nerves_hub_web/controllers/api/device_controller.ex:16]: Did you mean to remove these actions from being validated?

Comment (oestrich) [reply lib/nerves_hub_web/controllers/api/device_controller.ex:16]: Yeah, they’re validated in the action now, after device lookup

File: lib/nerves_hub_web/api_router.ex [hunk 2]
```diff
@@ -35,6 +35,17 @@ defmodule NervesHubWeb.APIRouter do
     post("/login", UserController, :login)
   end
 
+  scope "/devices", NervesHubWeb.API do
+    pipe_through([:api, :user])
```
Comment (jjcarstens) [inline lib/nerves_hub_web/api_router.ex:39]: Do we need the `:device` pipe still used here?

Comment (oestrich) [reply lib/nerves_hub_web/api_router.ex:39]: No, that loaded the device from the org in another plug. This undoes the plugs we have in place to be in the controller. I think once we’re further along in the migration I want to do we can replace the plugs with what all is going on here (maybe :shrug:)

**Review by jjcarstens — APPROVED** [review]

**Comment by oestrich** [comment]
Merged locally

**Pull request closed without merging**